### ISSUE TITLE
Does it work on the web

### ISSUE THREAD

**Issue by synw** [centrifugal/centrifuge-dart#34]
I tried to run the chat example on the web but it actually does not want to connect. It just hangs on the connection step, no output

**Comment by FZambia** [comment]
I never tried to use it in a web environment, so I think the answer is no. But I don't have an answer why - need some investigation. Maybe server logs has sth useful?

Could you give instructions on how to run example in a web context?

**Comment by mogol** [comment]
@FZambia the package depends on [dart:io](https://api.dart.dev/stable/2.7.2/dart-io/dart-io-library.html) which doesn't not work for web. 

Web package can be created with [ dart:html](https://api.dart.dev/stable/2.7.2/dart-html/dart-html-library.html). I do not see why it should not work, but some migration required :)

**Comment by Holofox** [comment]
@mogol, It would be nice to see Web support if you or someone else has some free time for this.

**Comment by synw** [comment]
@FZambia : to compile for the web: `flutter build web`. It could be nice to have an example online if it works on the web

**Comment by synw** [comment]
I tried to replace dart:io with [HtmlWebSocketChannel](https://pub.dev/packages/web_socket_channel#htmlwebsocketchannel) : [here is a gist](https://gist.github.com/synw/0925d2e5fcf077c4d3132db7614a2136) of `transport.dart`. It does the same thing, hanging at the connection step. The server says this (using the chat example):

```bash
error decoding command client=92fb9052-d0ca-4357-be6e-085542a42076 data=120,8,1,16,0,26,114,10,112,101,121,74,48,101,88,65,105,79,105,74,75,86,49,81,105,76,67,74,104,98,71,99,105,79,105,74,73,85,122,73,49,78,105,74,57,46,101,121,74,122,100,87,73,105,79,105,74,48,90,88,78,48,99,51,86,112,100,71,86,102,97,110,100,48,73,110,48,46,104,80,109,72,115,86,113,118,116,89,56,56,80,118,75,52,69,109,74,108,99,100,119,78,117,75,70,117,121,51,66,71,97,70,55,100,77,97,75,100,80,108,119 error="unexpected EOF" user=
```

I probably did not manage to use `HtmlWebSocketChannel` right: the params are not the same as dart:io and I am a bit lost on how to arrange this. It could help if someone with a good knowledge of this lib internals could check the gist

**Comment by FZambia** [comment]
Usually when working with binary data there is a method to turn WebSocket connection into binary mode. When I run an example I see that it sends text data instead of binary to server:
<img width="1062" alt="Screenshot 2020-04-18 at 18 39 58" src="https://user-images.githubusercontent.com/1196565/79642093-36aabc00-81a4-11ea-880c-7367591e147d.png">

So I think there should be a way to send binary.

**Comment by FZambia** [comment]
@synw this does the work:
```dart
_socket.sink.add(Uint8List.fromList(data));
```
I think similar conversion can be required for decoding replies from server too (not sure though).

**Comment by PlugFox** [comment]
You can use conditional imports for this issue
Usage example: https://github.com/PlugFox/platform_info/blob/master/lib/src/platform.dart#L10-L14

**Comment by FZambia** [comment]
We currently have an unfinished pr by @r-oman-G - https://github.com/centrifugal/centrifuge-dart/pull/53 - if someone interested to help it's a good starting point I suppose.

**Comment by robert-virkus** [comment]
Thanks to the PR of [r-oman-G](https://github.com/r-oman-G) I was able to implement  web support using [web_socket_channel](https://pub.dev/packages/web_socket_channel) , compare https://github.com/centrifugal/centrifuge-dart/pull/73

**Comment by FZambia** [comment]
Closed by #73, released as part of [v0.9.3](https://github.com/centrifugal/centrifuge-dart/releases/tag/v0.9.3)
